---
title: "Project"
output:
  html_document:
    df_print: paged
---



#call varients for hatchery 
```{bash}
for i in {01..05}; do
  /home/shared/bcftools-1.14/bcftools mpileup -Ou -f \
  /home/shared/8TB_HDD_02/thielkla/Karina-chinook/Olurida_v081.fa \
  /home/shared/8TB_HDD_02/thielkla/Karina-chinook/CSBM18H/CSMB18H.${i}.bam \
  | /home/shared/bcftools-1.14/bcftools call -mv -Ov \
  -o output/H${i}.vcf
done

```

#call varients for wildtype
```{bash}
for i in {01..05}; do
  /home/shared/bcftools-1.14/bcftools mpileup -Ou -f \
  /home/shared/8TB_HDD_02/thielkla/Karina-chinook/Olurida_v081.fa \
  /home/shared/8TB_HDD_02/thielkla/Karina-chinook/CSMB17W/CSMB17W.${i}.bam \
  | /home/shared/bcftools-1.14/bcftools call -mv -Ov -o \
  output/W${i}.vcf
done

```

# Merge so there can be a comparison 

```{bash}
for f in output/*.vcf; do
  /home/shared/htslib-1.14/bgzip "$f"         # compress to .vcf.gz
  /home/shared/htslib-1.14/tabix -p vcf "$f.gz"  # index for random access
done
```


```{bash}
ls output/*.vcf.gz > output/vcflist.txt
```


```{bash}
/home/shared/bcftools-1.14/bcftools merge \
-l output/vcflist.txt -Oz -o output/merged.vcf.gz

/home/shared/bcftools-1.14/bcftools index output/merged.vcf.gz
```


```{bash}
/home/shared/bcftools-1.14/bcftools query -l output/merged.vcf.gz
```

```{bash}
/home/shared/vcftools-0.1.16/bin/vcftools --gzvcf output/merged.vcf.gz \
  --weir-fst-pop hatchery.txt \
  --weir-fst-pop wild.txt \
  --out output/hatchery_vs_wild
```
ðŸ§  Interpretation
	â€¢	A mean FST near zero (or negative) suggests little to no genetic differentiation between the hatchery and wild populations.
	â€¢	Some negative FST values are possible due to sampling noise or low allele frequency variance; they are usually interpreted as zero.
	â€¢	A few high FST values (up to 1) indicate loci with strong differentiation â€” potentially under selection or drift.
	
	
```{r}
# Install if not already installed
#install.packages("DT")

# Load the package
#library(DT)

# Read your CSV file
fst_data <- read.csv("output/Top_1__FST_Loci.csv")

# Display interactive table
datatable(fst_data, options = list(pageLength = 10, scrollX = TRUE), rownames = FALSE)
```
	
	
