---
title: "Wildtype vs Hatchery o.lurida"
format: revealjs
editor: visual
---

## 

## Olympia oyster wildtype sequences vs hatchery in Central Sound \`\`\`{r setup, include=FALSE}

```{r setup, include=FALSE}
knitr::opts_chunk$set(
  echo = TRUE,         # Display code chunks
  eval = FALSE,         # Evaluate code chunks
  warning = FALSE,     # Hide warnings
  message = FALSE,     # Hide messages
  fig.width = 6,       # Set plot width in inches
  fig.height = 4,      # Set plot height in inches
  fig.align = "center", # Align plots to the center
  comment = ""         # Prevents appending '##' to beginning of lines in code output
)

```
## call varients for hatchery 

```{r, engine='bash'}
for i in {01..05}; do
  /home/shared/bcftools-1.14/bcftools mpileup -Ou -f \
  /home/shared/8TB_HDD_02/thielkla/Karina-chinook/Olurida_v081.fa \
  /home/shared/8TB_HDD_02/thielkla/Karina-chinook/CSBM18H/CSMB18H.${i}.bam \
  | /home/shared/bcftools-1.14/bcftools call -mv -Ov \
  -o ~/Karina-chinook/output/H${i}.vcf
done

```
## call varients for wildtype


```{r, engine='bash'}
for i in {01..05}; do
  /home/shared/bcftools-1.14/bcftools mpileup -Ou -f \
  /home/shared/8TB_HDD_02/thielkla/Karina-chinook/Olurida_v081.fa \
  /home/shared/8TB_HDD_02/thielkla/Karina-chinook/CSMB17W/CSMB17W.${i}.bam \
  | /home/shared/bcftools-1.14/bcftools call -mv -Ov \
  -o ~/Karina-chinook/output/W${i}.vcf
done
```
## Merge so there can be a comparison 

```{r, engine='bash'}
for f in ~/Karina-chinook/output/*.vcf; do
  /home/shared/htslib-1.14/bgzip "$f"         # compress to .vcf.gz
  /home/shared/htslib-1.14/tabix -p vcf "$f.gz"  # index for random access
done
```


```{r, engine='bash'}
ls ~/Karina-chinook/output/*.vcf.gz > ~/Karina-chinook/output/vcflist.txt
```


```{r, engine='bash'}
/home/shared/bcftools-1.14/bcftools merge \
-l ~/Karina-chinook/output/vcflist.txt -Oz -o ~/Karina-chinook/output/merged.vcf.gz

/home/shared/bcftools-1.14/bcftools index ~/Karina-chinook/output/merged.vcf.gz
```


```{r, engine='bash'}
/home/shared/bcftools-1.14/bcftools merge \
-l ~/Karina-chinook/output/vcflist.txt -Oz -o ~/Karina-chinook/output/merged.vcf.gz

/home/shared/bcftools-1.14/bcftools index ~/Karina-chinook/output/merged.vcf.gz
```

##Compare FST
#Create text file that list samples  

```{r, engine='bash'}
/home/shared/vcftools-0.1.16/bin/vcftools --gzvcf ~/Karina-chinook/output/merged.vcf.gz \
  --weir-fst-pop hatchery.txt \
  --weir-fst-pop wild.txt \
  --out ~/Karina-chinook/output/hatchery_vs_wild
```


```{r, engine='bash', eval=TRUE}
head -20 ~/Karina-chinook/output/hatchery_vs_wild.weir.fst 
```

## ðŸ§  Interpretation
	â€¢	A mean FST near zero (or negative) suggests little to no genetic differentiation between the hatchery and wild populations.
	
	
	â€¢	Some negative FST values are possible due to sampling noise or low allele frequency variance; they are usually interpreted as zero.
	
	
	â€¢	A few high FST values (up to 1) indicate loci with strong differentiation â€” potentially under selection or drift.
	
```{r, eval=FALSE}
# Install if not already installed
#install.packages("DT")

# Load the package
library(DT)

# Read the tab-delimited FST data
hatchery_vs_wild.weir.fst <- read.table("/home/shared/8TB_HDD_02/thielkla/Karina-chinook/output/hatchery_vs_wild.weir.fst", header = TRUE, sep = "\t", stringsAsFactors = FALSE)


# Display interactive table
datatable(hatchery_vs_wild.weir.fst, options = list(pageLength = 10, scrollX = TRUE), rownames = FALSE)
```
	
